pkg/controller/upload-controller.go

```diff
@@ -254,6 +320,48 @@ func (c *UploadController) initCerts() error {
 	return nil
 }
 
+func (c *UploadController) createCDIConfig() error {
+
+	var url string
+	routes, err := c.routeLister.List(labels.NewSelector())
+	if err != nil {
+		return err
+	}
+	for _, route := range routes {
+		url = getURLFromRoute(route)
+		if url != "" {
+			break
+		}
+	}
+	namespaces, err := getAllSupportedNamespaces(c.client)
+	if err != nil {
+		return err
+	}
+	if url == "" {
+		for _, namespace := range namespaces {
+			ingresses, err := c.client.ExtensionsV1beta1().Ingresses(namespace).List(metav1.ListOptions{
```

Comment:
Isn't it possible to have multiple ingresses? For different namespaces. This code takes the first it finds and finishes. Shouldn't we pass a list of ingress URLs to CreateCDIConfig with all the results found, or create one config for each namespace?

Reply:
Assuming CDI is only running in a single namespace, there should not be ingresses in any other namespaces.  There may be multiple in the same namespace though but prob unlikely.